# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/nsIHTMLEditor.idl
# Commit: 554d63dad12e
# Full Hash: 554d63dad12e4d1f2560e99de1fafac9a97da026
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2023-09-27 09:02:41
# Regressor Bug: 1851951
# File Overlap Count: 1
# Description:
#   Bug 1851951 - Make `HTMLEditor` refer computed `display` instead of the HTML default style at considering block or inline element r=m_kato
#   
#   Blink [1] and WebKit [2] refers `display-outside` value when they consider whether an
#   element is a block or an inline.
#   
# ==============================================================================

diff -r 97dc7be3b93c -r 554d63dad12e editor/nsIHTMLEditor.idl
--- a/editor/nsIHTMLEditor.idl	Wed Sep 27 02:50:33 2023 +0000
+++ b/editor/nsIHTMLEditor.idl	Wed Sep 27 03:01:26 2023 +0000
@@ -93,12 +93,14 @@
   /* ------------ HTML content methods -------------- */
 
   /**
-   * Tests if a node is a BLOCK element according the the HTML 4.0 DTD.
-   *   This does NOT consider CSS effect on display type
+   * Tests if a node is a BLOCK element.  It's depend on
+   * `editor.block_inline_check.use_computed_style` pref whether this refers
+   * the computed style or the default style.
    *
    * @param aNode      the node to test
    */
-  boolean nodeIsBlock(in Node node);
+  [can_run_script]
+  boolean nodeIsBlock(in Node aNode);
 
   /**
    * Insert some HTML source at the current location